Mage.Sets/src/mage/cards/r/RatadrabikOfUrborg.java

```diff
@@ -0,0 +1,110 @@
+package mage.cards.r;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.ObjectColor;
+import mage.abilities.Ability;
+import mage.abilities.common.DiesCreatureTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenCopyTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAllEffect;
+import mage.constants.*;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.keyword.WardAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author rockydirtbag
+ */
+public final class RatadrabikOfUrborg extends CardImpl {
+
+    private static final FilterControlledCreaturePermanent filter = new FilterControlledCreaturePermanent("Zombies you control");
+
+    private static final FilterControlledCreaturePermanent filter2 = new FilterControlledCreaturePermanent("another legendary creature you control");
+
+    static {
+        filter.add(SubType.ZOMBIE.getPredicate());
+        filter2.add(SuperType.LEGENDARY.getPredicate());
+        filter2.add(AnotherPredicate.instance);
+    }
+
+    public RatadrabikOfUrborg(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{W}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ZOMBIE);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Ward {2}
+        this.addAbility(new WardAbility(new ManaCostsImpl<>("{2}")));
+
+        // Other zombies you control have vigilance.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityAllEffect(
+                VigilanceAbility.getInstance(), Duration.WhileOnBattlefield, filter, true
+        )));
+
+        // Whenever another legendary creature you control dies, create a token that's a copy of that creature,
+        // except it's not legendary and it's a 2/2 black Zombie in addition to its other colors and types.
+        this.addAbility(new DiesCreatureTriggeredAbility(new RatadrabikOfUrborgEffect(),false, filter2, true));
+
+    }
+
+    private RatadrabikOfUrborg(final RatadrabikOfUrborg card) {
+        super(card);
+    }
+
+    @Override
+    public RatadrabikOfUrborg copy() {
+        return new RatadrabikOfUrborg(this);
+    }
+}
+
+class RatadrabikOfUrborgEffect extends OneShotEffect {
+
+    public RatadrabikOfUrborgEffect() {
+        super(Outcome.Benefit);
```

Comment:
I think this should be `.PutCreatureIntoPlay` or `.Copy`

Reply:
Great, thanks for your help.
There is still a lot of guessing while I'm coding in xMage... Although I've been looking at a lot of cards, I'm still not sure what Outcome does exactly or where it matters. Any advice on this?

Reply:
In theory, Outcome is used by the AI, but I don't think most of the specific ones matter.